woocommerce/woocommerce
# Update task list items to show spinner on selection

### All Submissions:




### Changes proposed in this Pull Request:





### Summary:
This pull request introduces a spinner component to indicate the loading status when the user clicks the product task.

### Reason for the change:
When clicking on the product tasks, there is a noticeable delay before being taken to the next page.
Previously, users were left waiting without any indication of the loading status.

### Steps to fix:

This PR introduces a spinner component to indicate the loading status during the async actions on clicking the product task. Although the loading time optimization is not included in this PR, the spinner serves as an initial improvement to provide visual feedback to users during the loading process.

Original behaviour - No Spinner shows on task selection

![No Spinner before](https://github.com/woocommerce/woocommerce/assets/21078987/776f202c-ac6d-465f-8538-6684cb3d5177)

After fix - Spinner displays showing loading status on selection

![Spinner shows on selection](https://github.com/woocommerce/woocommerce/assets/21078987/7f44ac0d-65e4-4c7e-a683-69ec6d3ed61b)


Closes https://github.com/woocommerce/woocommerce/issues/36510



### How to test the changes in this Pull Request:

Case 1: Show spinner




1. Pull down this branch on a new or existing site.
2. Go to WooCommerce > Home and click on `Add Products` to open the product task list.
3. Click on any product task (For example- Physical product) 
4. Ensure the spinner shows until the page redirects to another page. 
5. If the redirection is unsuccessful due to an error, the spinner is hidden.

https://github.com/woocommerce/woocommerce/assets/21078987/98ce0770-dcda-485e-af6a-93ec285f7fbf

Case 2: Hide spinner

If the page redirection fails on clicking the product task list item, we should hide the spinner so it doesn't show indefinitely. To test this case, I changed the variable's value to null while debugging the app to generate the error.

1. First, set up WooCommerce in the dev mode. Clone the repo from [this](https://github.com/woocommerce/woocommerce) link.
2. Run `nvm use` and `pnpm install` from plugins/woocommerce
3. Start the environment by running `pnpm -- wp-env start`
4. Go to plugins/woocommerce-admin and run `pnpm dev` to build the dev version, as we need to debug the code in the WooCommerce-admin.
5. Once the page loads, follow the steps in the video below to generate the error and ensure spinner is hidden.


https://github.com/woocommerce/woocommerce/assets/21078987/e28f4d72-1181-42f0-99bb-692a91dbc961
